grpc/src/main/java/io/stargate/grpc/codec/cql/CollectionCodec.java

```diff
@@ -0,0 +1,58 @@
+package io.stargate.grpc.codec.cql;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.Column.ColumnType;
+import io.stargate.proto.QueryOuterClass.Collection;
+import io.stargate.proto.QueryOuterClass.Value;
+import java.nio.ByteBuffer;
+
+public class CollectionCodec extends CompositeCodec {
+
+  @Override
+  public ByteBuffer encode(@NonNull Value value, @NonNull Column.ColumnType type) {
+    if (!value.hasCollection()) {
+      throw new IllegalArgumentException("Expected collection type");
+    }
+    assert type.isCollection();
+
+    Collection collection = value.getCollection();
+
+    Column.ColumnType elementType = type.parameters().get(0);
+    ValueCodec elementCodec = ValueCodecs.get(elementType.rawType());
+
+    int elementCount = collection.getElementsCount();
+    ByteBuffer[] encodedElements = new ByteBuffer[elementCount];
+    int toAllocate = 4;
```

Comment:
What's this?

Reply:
Space for the number of elements in the collection.

Reply:
https://github.com/stargate/stargate/pull/936/files/bff0e34c80a4cdcfcd34c2b07e6a6fd11b8c7533#diff-8379a2c1845030448fcd49119bf131681bb724d112e78d0974cc91c837abc622R39